### ISSUE TITLE
Xib support

### ISSUE THREAD

**Issue by lemonmojo** [Microsoft/WinObjC#27]
I have created a very basic Xcode project with a single Xib file containing a UIView with several text fields and a button. I'm not using Auto Layout or Storyboards.
In my AppDelegate I'm instantiating the corresponding View Controller and add it to my Navigation Controller.

This works fine on iOS but the only thing I get when importing the project into Visual Studio is the Navigation Controller's title - everything else is black when running on Windows 10.
I have verified that my View Controller at least gets loaded by putting a breakpoint in the viewDidLoad method which indeed gets hit.

The WOCCatalog sample works fine but I have noticed there are no Xibs included in this project.

So my question is, are Xibs (even with limitations, like no Auto Layout) supported at this stage?

**Comment by SquaredTiki** [comment]
Looking at the `xib2nib` tool it seems to be using the old XIB format (pre-version 3) for some reason for all XIBs but then the correct new format for Storyboards. Not sure if this is related.

**Comment by lemonmojo** [comment]
@SquaredTiki Yep, the problem seems to be related to the Xib2Nib tool. I just tried invoking it from the command line and it crashes instantly. Unfortunately there's no error message coming along with the crash.

**Comment by SquaredTiki** [comment]
@lemonmojo Probably as a result of it looking for a key that doesn't exist. All the code needed appears to be there to make it work with the 'new' (couple of years old) XIB version as it is exactly the same format that Storyboard's use, however it hasn't been implemented.

**Comment by alexdrenea** [comment]
Are there any updates in supporting Storyboards / Xibs?
All the threads regarding this seem to be really old. The roadmap states that there is "preliminary Storyboard support" and "preliminary xib support". What does this mean ?

Thank you.

**Comment by technotrance** [comment]
On board support to stop viruses and hacks by diplaying what kind of trojan was implemented and my repository repairs all the above and would be in the library core drawer its asking for action to be taken by showind you the diagnostics visual on your screen and programme

**Comment by nicholasgerard-msft** [comment]
@alexdrenea vsimporter currently provides support for Storyboards and Xibs using xib2nib. It doesn't support *all* Storyboard/Xib features, but many standard use cases are supported and should convert without issue. It's worth noting, however, that Visual Studio does not currently have a Storyboard editor, so if you build your layout using Storyboards, any changes you make will have to be made on the Xcode side and then run through vsimporter again. We're also working on an updated and more capable version of xib2nib.

**Comment by alexdrenea** [comment]
Thanks for the note.
My storyboard looks like was converted (no error from xib2nib) but when I run the app, I am getting a white screen (the viewDidLoad does not fire in the viewcontroller). It's what I think to be a standard storyboard with a few elements on the screen (I am using AutoLayout though).
Do you think this comes from the storyboard or should I look somewhere else?
Thanks,

**Comment by nicholasgerard-msft** [comment]
@alexdrenea When you run vsimporter, do you get any error messages? And when you run your app, can you provide any of the debug output?

**Comment by alexdrenea** [comment]
I didn't get any errors or warnings when I ran vsimporter.
I was able to clone the repo, compile the sources and debug my app startup. 
When app is starting up, in UIApplicationMain.mm - UIApplicationMainInit it gets to a point where it tries to initialize the Storyboard ( UIStoryboard* storyBoard = [UIStoryboard storyboardWithName:storyBoardName bundle:[NSBundle mainBundle]]; ) and this returns Nil which means the rootController does not get set.
